### PR TITLE
Fix Tests

### PR DESCRIPTION
**Pull request by dahlbyk**
Following on from #878, trying to get back to all green.

- [ ] **Correctly reverts the Window Title back to original state** is unreliable; might just skip
- [x] `Expected: 'igf ' But was:  'git checkout '` seems to be a change in how either Pester or PowerShell handles `-Scope Script`; changing to `-Scope Global` seems to fix. (@dscho you were close!)
- [x] **Tab completes add file in working dir with special char as quoted** suggests `core.quotepath=false` (from b689b5e2974ef8365e16108eb4f7e0edcccfdab6 in #403) no longer affects `git status -sb` as expected - a path with special characters is coming back in `""` regardless. This feels like a regression, but I haven't gone looking for release notes.

### CODE REVIEW

**Comment by dahlbyk** [comment]
> * a path with special characters is coming back in `""` regardless

Turns out it's only spaces? https://github.com/git/git/commit/f3fc4a1b8680c114defd98ce6f2429f8946a5dc1

**Comment by dahlbyk** [comment]
> > * a path with special characters is coming back in `""` regardless
> 
> Turns out it's only spaces? [git/git@f3fc4a1](https://github.com/git/git/commit/f3fc4a1b8680c114defd98ce6f2429f8946a5dc1)

I'm genuinely confused how this test ever passed. https://github.com/git/git/commit/dbfdc625a5aad10c47e3ffa446d0b92e341a7b44 is from 2010. 🤔

File: src/GitUtils.ps1 [hunk 1]
```diff
@@ -334,20 +334,26 @@ function Get-GitStatus {
                     '^(?<index>[^#])(?<working>.) (?<path1>.*?)(?: -> (?<path2>.*))?$' {
                         if ($sw) { dbg "Status: $_" $sw }
 
+                        $path1 = $matches['path1']
+
+                        if ($path1 -like '"*"') {
+                            $path1 = $path1.Substring(1, $path1.Length - 2)
```
Comment (rkeithhill) [inline src/GitUtils.ps1:340]: A comment here on why the path is being adjusted would other maintainers.  :-)

File: test/DefaultPrompt.Tests.ps1 [hunk 2]
```diff
@@ -531,17 +533,12 @@ M test/Baz.Tests.ps1
         Context 'Removing the posh-git module' {
             It 'Correctly reverts the Window Title back to original state' {
                 Set-Item function:\prompt -Value ([Runspace]::DefaultRunspace.InitialSessionState.Commands['prompt']).Definition
+                $originalTitle = & $module Get-Variable OriginalWindowTitle -ValueOnly
```
Comment (rkeithhill) [inline test/DefaultPrompt.Tests.ps1:536]: Why not?
```
$originalTitle = & $module { $OriginalWindowTitle }
```

Comment (rkeithhill) [reply test/DefaultPrompt.Tests.ps1:536]: But do it after importing posh-git

File: test/DefaultPrompt.Tests.ps1 [hunk 3]
```diff
@@ -531,17 +533,12 @@ M test/Baz.Tests.ps1
...
-                if ($Host.Name -eq 'RemoteHostImplementation') {
-                    $title | Should -eq $originalTitle
-                }
-                elseif ($PSVersionTable.PSVersion.Major -lt 6) {
-                    $title | Should -match '^Windows PowerShell|:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe$'
-                }
-                else {
-                    $title | Should -match '^(Administrator: )?(Windows )?PowerShell'
-                }
+                $title | Should -eq $originalTitle
```
Comment (rkeithhill) [inline test/DefaultPrompt.Tests.ps1:541]: Yeah, that's better.  😁

File: test/TabExpansion.Tests.ps1 [hunk 4]
```diff
@@ -357,7 +358,7 @@ Describe 'TabExpansion Tests' {
             $result | Should -BeExactly "'v2.0.0'''"
         }
         It 'Tab completes add file in working dir with special char as quoted' {
-            $filename = 'foo{bar} (x86).txt';
+            $filename = 'foo{bar} ��(x86).txt';
```
Comment (rkeithhill) [inline test/TabExpansion.Tests.ps1:361]: What's with the unrecognized chars?  Is that part of the "special char" test?  Might be worth adding as another test for emoji chars or whatever that is.  IIRC this test was meant to check for tab completion with PowerShell and/or regex special chars/syntax e.g. `{}` and `()` in the completed text.

**Review by rkeithhill — APPROVED** [review]
Just a few suggestions otherwise LGTM

File: test/DefaultPrompt.Tests.ps1 (hunk 2 above)
Comment (rkeithhill) [reply test/DefaultPrompt.Tests.ps1:536]: Ah, now I see you're capturing this variable in the test fixture before the test runs.

File: test/TabExpansion.Tests.ps1 (hunk 4 above)
Comment (dahlbyk) [reply test/TabExpansion.Tests.ps1:361]: Was just being lazy, but yeah the emoji is to ensure we're setting `core.quotePath` (and it's working as expected). Will break out as a separate test.

File: test/DefaultPrompt.Tests.ps1 (hunk 2 above)
Comment (dahlbyk) [reply test/DefaultPrompt.Tests.ps1:536]: > Why not?
> 
> ```
> $originalTitle = & $module { $OriginalWindowTitle }
> ```

I don't use this syntax often, so I refer to https://github.com/dahlbyk/posh-git/blob/5a87e8a196d13395f978d774f80dedb40d43b85e/src/Utils.ps1#L7

File: test/TabExpansion.Tests.ps1 (hunk 4 above)
Comment (dahlbyk) [reply test/TabExpansion.Tests.ps1:361]: > > What's with the unrecognized chars? Is that part of the "special char" test? Might be worth adding as another test for emoji/unprintable chars or whatever that is. IIRC this test was meant to check for tab completion with PowerShell and/or regex special chars/syntax e.g. `{}` and `()` in the completed text.
>
> Was just being lazy, but yeah the emoji is to ensure we're setting `core.quotePath` (and it's working as expected). Will break out as a separate test.

Turns out we already had a test for this: https://github.com/dahlbyk/posh-git/blob/a62f90cb1ff1b66bc05da5787eb561bce64e3da6/test/TabExpansion.Tests.ps1#L222-L226

Relocated into the appropriate `Context`.

**Review by dahlbyk — COMMENTED** [review]
Thanks for the review, @rkeithhill

**Pull request merged**